Products/CMFPlone/browser/syndication/settings.py

```diff
@@ -17,19 +17,30 @@ class FeedSettings(object):
 
     def __init__(self, context):
         self.context = context
-        annotations = IAnnotations(context)
+        self.annotations = IAnnotations(context)
+        self.needs_saving = False
 
-        self._metadata = annotations.get(FEED_SETTINGS_KEY, None)
+        self._metadata = self.annotations.get(FEED_SETTINGS_KEY, None)
         if self._metadata is None:
             self._metadata = PersistentDict()
-            annotations[FEED_SETTINGS_KEY] = self._metadata
+            self.needs_saving = True
 
         registry = getUtility(IRegistry)
         self.site_settings = registry.forInterface(ISiteSyndicationSettings,
                                                    check=False)
 
+    def _set(self):
+        """
+        what are we doing here you might ask?
+        well, this causes us to write on read so only set on annotation
+        if we need to
+        """
+        if self.needs_saving:
+            self.annotations[FEED_SETTINGS_KEY] = self._metadata
+
     def __setattr__(self, name, value):
-        if name in ('context', '_metadata', 'site_settings'):
+        if name in ('context', '_metadata', 'site_settings', 'annotations',
+                    'needs_saving'):
             self.__dict__[name] = value
         else:
             self._metadata[name] = value
```

Comment:
You missed just one part of the back-port. need to call self._set()

https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/browser/syndication/settings.py#L47